### PR TITLE
fix(concentrator): fix abc token fetcher

### PR DESCRIPTION
**Pull request by 4E-69-63-6B-20-42-6F-73**
## Description

Moved the abcTokenFetcher from Clever to Concentrator and implemented the missing getPrice and getLiquidity functions.

getPrice uses the concentrator API, because there is no dex pool for it, is this function cached?

## Checklist

- [X] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] (optional) As a contributor, my Ethereum address/ENS is:
- [ ] (optional) As a contributor, my Twitter handle is:

### CODE REVIEW

File: src/apps/concentrator/ethereum/concentrator.abc.token-fetcher.ts [hunk 1]
```diff
@@ -35,7 +34,16 @@ export class EthereumCleverAbcTokenFetcher extends AppTokenTemplatePositionFetch
...
   }
 
-  async getPricePerShare() {
-    return [(await this.getContract(abcCVX).ratio()).div(10 ** 6).toNumber() / 2];
+  async getPricePerShare({ } : GetPricePerShareParams<AbcCvx> )  {
+    return [1]; 
+  }
+
+  async getPrice({ }: GetPriceParams<AbcCvx>): Promise<number> {
+    return await axios.get('https://api.aladdin.club/api/aladdin/initInfo').then(v => v.data.data.abcCVXPrice)
```
Comment (wpoulin) [inline src/apps/concentrator/ethereum/concentrator.abc.token-fetcher.ts:42]: You shouldn't have to fetch prices from an external sources

File: src/apps/concentrator/ethereum/concentrator.abc.token-fetcher.ts [hunk 2]
```diff
@@ -35,7 +34,16 @@ export class EthereumCleverAbcTokenFetcher extends AppTokenTemplatePositionFetch
     ];
   }
 
-  async getPricePerShare() {
-    return [(await this.getContract(abcCVX).ratio()).div(10 ** 6).toNumber() / 2];
+  async getPricePerShare({ } : GetPricePerShareParams<AbcCvx> )  {
+    return [1];
```
Comment (wpoulin) [inline src/apps/concentrator/ethereum/concentrator.abc.token-fetcher.ts:38]: You can't have 2 underlying tokens and only 1 price per share value. Plus, it's not true the pricePerShare between this token and the underlying is 1 to 1.

Here's the formula for the pricePerShare 
`pricePerShare = reserves.map(r => r/supply)`

**Review by wpoulin — CHANGES_REQUESTED** [review]

File: src/apps/concentrator/ethereum/concentrator.abc.token-fetcher.ts (hunk 1 above)
Comment (4E-69-63-6B-20-42-6F-73) [reply src/apps/concentrator/ethereum/concentrator.abc.token-fetcher.ts:42]: Okay, but how should I get it then? Since there is no dex pool or something

Comment (4E-69-63-6B-20-42-6F-73) [reply src/apps/concentrator/ethereum/concentrator.abc.token-fetcher.ts:42]: O I see, the price will be calculated based on pricePerShare it it isn't overridden.

File: src/apps/concentrator/ethereum/concentrator.abc.token-fetcher.ts (hunk 2 above)
Comment (4E-69-63-6B-20-42-6F-73) [reply src/apps/concentrator/ethereum/concentrator.abc.token-fetcher.ts]: Done!

File: src/apps/concentrator/ethereum/concentrator.abc.token-fetcher.ts (hunk 1 above)
Comment (wpoulin) [reply src/apps/concentrator/ethereum/concentrator.abc.token-fetcher.ts]: All you need to do is fetch the underlying reserves, apply the formula below and bingo !

```
  async getUnderlyingTokenDefinitions(/*{ contract }: GetUnderlyingTokensParams<AbcCvx>*/) {
    return [
      { address: '0xf9078fb962a7d13f55d40d49c8aa6472abd1a5a6', network: this.network },
      { address: '0xf05e58fcea29ab4da01a495140b349f8410ba904', network: this.network },
    ];
  }

  async getPricePerShare({ appToken, contract }: GetPricePerShareParams<AbcCvx>) {
    const reserves = await Promise.all([contract.totalCurveLpToken(), contract.totalDebtToken()]);

    const pricePerShareRaw = reserves.map(r => Number(r) / appToken.supply);
    const pricePerShare = pricePerShareRaw.map(x => x / 10 ** appToken.decimals);

    return pricePerShare;
  }
```

**Review by wpoulin — APPROVED** [review]

**Pull request merged**